### PR TITLE
CONSOLE-2843, CONSOLE-3394: Add ManagedProxyServiceResolver for thanos-querier on all managed clusters

### PR DESCRIPTION
**Pull request by TheRealJon**

### CODE REVIEW

File: pkg/console/controllers/thanosquerierproxyserviceresolver/controller.go
Comment (jhadvig) [inline pkg/console/controllers/thanosquerierproxyserviceresolver/controller.go:156]: nit. just to stay consistant with the other remove methods that our controllers have
```suggestion
func (c *ThanosQuerierProxyResolverController) removeManagedProxyServiceResolver(ctx context.Context) error {
```

File: pkg/console/controllers/thanosquerierproxyserviceresolver/controller.go
Comment (jhadvig) [inline pkg/console/controllers/thanosquerierproxyserviceresolver/controller.go:63]: not sure why config informer is needed here

File: pkg/console/subresource/managedserviceproxyresolver/thanos-querier.go
Comment (jhadvig) [inline pkg/console/subresource/managedserviceproxyresolver/thanos-querier.go:14]: so this looks like a static manifest, lets put it into the bindata

File: pkg/console/subresource/managedserviceproxyresolver/thanos-querier.go
Comment (TheRealJon) [reply pkg/console/subresource/managedserviceproxyresolver/thanos-querier.go:20]: Yeah. I don't know the details, but essentially you have to create this resource in order for the cluster proxy to resolve requests to a managed cluster service.

File: pkg/console/subresource/managedserviceproxyresolver/thanos-querier.go
Comment (TheRealJon) [reply pkg/console/subresource/managedserviceproxyresolver/thanos-querier.go:14]: The problem is we don't have a 'ReadOrDie' func for ManagedProxyServiceResolver resources.

**Comment by TheRealJon** [comment]
@jhadvig I've made some changes if you don't mind reviewing again

File: pkg/console/subresource/util/util.go
Comment (TheRealJon) [reply pkg/console/subresource/util/util.go:138]: I ran that, but didn't see any function like this generated

File: pkg/console/controllers/thanosquerierproxyserviceresolver/controller.go
Comment (jhadvig) [inline pkg/console/controllers/thanosquerierproxyserviceresolver/controller.go:1]: this pkg is the same as `pkg/console/controllers/advancedclustermanagement/thanosquerierproxyserviceresolver/controller.go`

File: pkg/console/subresource/managedserviceproxyresolver/apply.go
Comment (jhadvig) [inline pkg/console/subresource/managedserviceproxyresolver/apply.go:21]:
```suggestion
			klog.V(4).Infof("failed to create thanos-querier ManagedServiceProxyResolver: %s/n", err)
```

File: pkg/console/subresource/managedserviceproxyresolver/apply.go
Comment (jhadvig) [inline pkg/console/subresource/managedserviceproxyresolver/apply.go:19]: If we need to create the resource, we either need. to keep it or get it again, otherwise the variable will contain `nil`
```suggestion
		existing, err = client.Create(ctx, required, metav1.CreateOptions{})
```

File: pkg/console/subresource/managedserviceproxyresolver/apply.go
Comment (jhadvig) [inline pkg/console/subresource/managedserviceproxyresolver/apply.go:24]: return will trigger another loop. we could just continue if no error occurs

File: pkg/console/subresource/util/util.go
Comment (jhadvig) [reply pkg/console/subresource/util/util.go:138]: hmmm yeah I see that ... probably not all kinds can be generated

**Review by jhadvig — CHANGES_REQUESTED** [review]

File: pkg/console/subresource/managedserviceproxyresolver/apply.go
Comment (TheRealJon) [reply pkg/console/subresource/managedserviceproxyresolver/apply.go:33]: I'm not sure I follow. This func only returns an error. If it returns nil, that means the required resource was successfully applied and no further action is required from the controller.

File: pkg/console/subresource/managedserviceproxyresolver/apply.go
Comment (TheRealJon) [reply pkg/console/subresource/managedserviceproxyresolver/apply.go:24]: We shouldn't need to do anything with the result of this create call. If successful, that means the required resource was created and we don't need to do anything more.

File: pkg/console/subresource/managedserviceproxyresolver/apply.go
Comment (jhadvig) [reply pkg/console/subresource/managedserviceproxyresolver/apply.go:33]: hmm looking at my comment now it seems I've already had christmas spirit, while writing 😄  please ignore

File: pkg/console/controllers/advancedclustermanagement/thanosquerierproxyserviceresolver/controller.go
Comment (jhadvig) [inline pkg/console/controllers/advancedclustermanagement/thanosquerierproxyserviceresolver/controller.go:99]:
```suggestion
		klog.V(4).Infoln("Skipping thanos-querier ManagedProxyServiceResolver sync. API is missing"))
```

File: pkg/console/controllers/advancedclustermanagement/thanosquerierproxyserviceresolver/controller.go
Comment (jhadvig) [inline pkg/console/controllers/advancedclustermanagement/thanosquerierproxyserviceresolver/controller.go:104]:
```suggestion
		klog.V(4).Infof("Error getting ManagedProxyServiceResolver CRD: %s/n", err))
```

File: pkg/console/controllers/advancedclustermanagement/thanosquerierproxyserviceresolver/controller.go
Comment (jhadvig) [inline pkg/console/controllers/advancedclustermanagement/thanosquerierproxyserviceresolver/controller.go:96]: nit
```suggestion
	// Confirm that the ManagedProxyServiceResolver API is installed before proceeding
```

File: pkg/console/subresource/managedserviceproxyresolver/apply.go
Comment (jhadvig) [inline pkg/console/subresource/managedserviceproxyresolver/apply.go:28]: nit
```suggestion
			klog.V(4).Infof("failed to create ManagedServiceProxyResolver '%s': %s/n", required.Name, err)
```

File: pkg/console/subresource/managedserviceproxyresolver/apply.go
Comment (jhadvig) [inline pkg/console/subresource/managedserviceproxyresolver/apply.go:38]: nit
```suggestion
		klog.V(4).Infof("failed to get ManagedServiceProxyResolver '%s': %s/n", required.Name, err)
```

File: pkg/console/subresource/managedserviceproxyresolver/apply.go
Comment (jhadvig) [inline pkg/console/subresource/managedserviceproxyresolver/apply.go:55]:
```suggestion
			klog.V(4).Infof("failed to update ManagedServiceProxyResolver '%s': %s/n", required.Name, err)
```

**Review by jhadvig — COMMENTED** [review]
Found couple of nits, otherwise the change looks good 👍  PTAL

**Comment by jhadvig** [comment]
/approved

QE Approver:
 /assign @yapei
Docs Approver:
 /assign @opayne1
PX Approver:
/assign @RickJWagner

**Review by opayne1 — COMMENTED** [review]
/label docs-approved

**Comment by jhadvig** [comment]
```
--- FAIL: TestProxy (246.83s)
```

**Comment by jhadvig** [comment]
```
clusteroperator/authentication is progressing: OAuthServerDeploymentProgressing: deployment/oauth-openshift.openshift-authentication: 1/3 pods have been updated to the latest generation
	clusteroperator/dns is progressing: DNS "default" reports Progressing=True: "Have 4 available node-resolver pods, want 6."
	clusteroperator/etcd is degraded because ClusterMemberControllerDegraded: unhealthy members found during reconciling members
EtcdEndpointsDegraded: EtcdEndpointsController can't evaluate whether quorum is safe: etcd cluster has quorum of 2 and 2 healthy members which is not fault tolerant: [{Member:ID:2977997594514885420 name:"ip-10-0-185-30.ec2.internal" peerURLs:"https://10.0.185.30:2380/" clientURLs:"https://10.0.185.30:2379/"  Healthy:true Took:1.031461ms Error:} {Member:ID:3927196257551455866 name:"ip-10-0-219-78.ec2.internal" peerURLs:"https://10.0.219.78:2380/" clientURLs:"https://10.0.219.78:2379/"  Healthy:false Took: Error:create client failure: failed to make etcd client for endpoints [https://10.0.219.78:2379/]: context deadline exceeded} {Member:ID:6348907777362893031 name:"ip-10-0-151-49.ec2.internal" peerURLs:"https://10.0.151.49:2380/" clientURLs:"https://10.0.151.49:2379/"  Healthy:true Took:1.432959ms Error:}]
EtcdMembersDegraded: 2 of 3 members are available, ip-10-0-219-78.ec2.internal is unhealthy
```

/retest

**Review by jhadvig — APPROVED** [review]
/lgtm
/retest

**Comment by spadgett** [comment]
Overriding known failing jobs to unblock the team. The failure is not related to this PR.

/override ci/prow/e2e-agnostic-upgrade

/retest

**Comment by yapei** [comment]
tested together with https://github.com/openshift/console/pull/12360, no issues found
/label qe-approved

**Comment by jhadvig** [comment]
Overriding since the tests already [passed](https://github.com/openshift/console-operator/pull/707#issuecomment-1420713875) but all tests have been triggered again, due to other's PR merge.
  
/override ci/prow/e2e-agnostic-upgrade
/overide ci/prow/e2e-aws-console

**Comment by jhadvig** [comment]
Overriding since the tests already https://github.com/openshift/console-operator/pull/707#issuecomment-1420713875 but all tests have been triggered again, due to other's PR merge.

/override ci/prow/e2e-agnostic-upgrade
/override ci/prow/e2e-aws-console

**Pull request merged**
